models/purchase_order.py

```diff
@@ -0,0 +1,29 @@
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+    _description = 'Purchase Order Enhancement'
+
+    active = fields.Boolean(string='Active', default=True)
+    lifespan = fields.Integer(string='Lifespan', default=30)
+    hotline = fields.Char(string='Hotline')
+
+    def action_archive_purchase_order(self):
+        if not self.env.user or self.env.user.user_has_groups('purchase.group_purchase_manager'):
```

Comment:
The condition ```not self.env.user``` here does not satisfy the rule _Only Purchase Manger can archive Purchase Order(s)_